src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -564,32 +548,56 @@ public HttpClientChannelAsyncRequest(HttpClientRequestChannel channel)
                         if (_channel.State != CommunicationState.Opened)
                         {
                             // if we were aborted while getting our request or doing correlation, 
-                            // we need to abort the web request and bail
+                            // we need to abort the request and bail
                             Cleanup();
                             _channel.ThrowIfDisposedOrNotOpen();
                         }
 
-                        HttpOutput httpOutput = HttpOutput.CreateHttpOutput(_httpRequestMessage, _factory, request, _factory.IsChannelBindingSupportEnabled, _factory.GetHttpClient(), _factory._authenticationScheme, _timeoutHelper);
+                        bool suppressEntityBody = PrepareMessageHeaders(request);
 
+                        if (!suppressEntityBody)
+                        {
+                            _httpRequestMessage.Content = MessageContent.Create(_factory, request, _timeoutHelper);
+                        }
+
+                        await SendPreauthenticationHeadRequestIfNeeded();
                         bool success = false;
+
                         try
                         {
-                            _httpResponseMessageTask = await httpOutput.SendAsync(_httpRequestMessage);
-                            //this.channelBinding = httpOutput.TakeChannelBinding();
-                            await httpOutput.CloseAsync();
+                            _httpResponseMessage = await _httpClient.SendAsync(_httpRequestMessage, HttpCompletionOption.ResponseHeadersRead, _timeoutHelper.CancellationToken);
+                            _httpRequestMessage.Dispose();
```

Comment:
It's the request message I'm calling `Dispose()` on here, not the response message. We don't want to null the field out just yet as we might still want to retrieve some properties from the object. Calling `Dispose()` causes the contained `HttpContent` object to be disposed, which in our case will return buffers to the `BufferManager` and dispose the contained `Stream`. If no exception has been thrown, that means the entire request stream has been sent to the server, and the HTTP response headers have been received. This means we don't need the request object any more. I'll add a short comment.